### ISSUE TITLE
config.guess (packaged xslt) broken on Ubuntu + FIX

### ISSUE THREAD

**Issue by e2** [sparklemotion/nokogiri#1402]
First, I didn't include full info, logs. Because ... lots of reasons. (But I'll include error output for search engines to lead people here).

#### Problem

`gem install nokogiri -v '1.6.7.1'` fails on Ubuntu.

#### Reason

Syntax error in config.guess.

#### Cause

The EOF isn't recognized as correctly ending the heredoc:

https://github.com/sparklemotion/nokogiri/blob/712af569d2f09562208e1c52121c47c047f52a5b/patches/libxslt/0017-Updated-config.guess.patch#L104

#### The fix:

https://github.com/chronomantic/nokogiri/commit/95247a38145e6bafabbd1d9a5bda0b9eb1e2cff8?diff=unified

**Comment by e2** [comment]
#### Relevant (truncated) output of `gem install ...`

Note: no useful hints here (since output of config.guess was garbage).

```
Running git apply with /cache/me/.rvm/gems/ruby-2.2.2/gems/nokogiri-1.6.7.1/patches/libxslt/0017-Updated-config.guess.patch... OK
Running 'configure' for libxslt 1.1.28... ERROR, review '/cache/me/.rvm/gems/ruby-2.2.2/gems/nokogiri-1.6.7.1/ext/nokogiri/tmp/x86_64-unknown-linux-gnu/ports/libxslt/1.1.28/configure.log' to see what happened. Last lines are:
========================================================================
#endif
#if defined (host_mips) && defined (MIPSEB)
#if defined (SYSTYPE_SYSV)
  printf ("mips-mips-riscos%ssysv\n", argv[1]); exit (0);
#endif
#if defined (SYSTYPE_SVR4)
  printf ("mips-mips-riscos%ssvr4\n", argv[1]); exit (0);
#endif
#if defined (SYSTYPE_BSD43) || defined(SYSTYPE_BSD)
  printf ("mips-mips-riscos%sbsd\n", argv[1]); exit (0);
#endif
#endif
  exit (-1);
}

/usr/lib/gcc/x86_64-linux-gnu/5/../../../x86_64-linux-gnu/crt1.o: In function `_start':
/build/buildd/glibc-2.21/csu/../sysdeps/x86_64/start.S:114: undefined reference to `main'
collect2: error: ld returned 1 exit status
Invalid configuration `mips-mips-riscos4.2.0-22-lowlatency': machine `mips-mips-riscos4.2.0-22' not recognized
configure: error: /bin/bash ./config.sub mips-mips-riscos4.2.0-22-lowlatency failed
========================================================================
*** extconf.rb failed ***
Could not create Makefile due to some reason, probably lack of necessary
libraries and/or headers.  Check the mkmf.log file for more details.  You may
need configuration options.
```

**Comment by e2** [comment]
#### Result from running `./configure` manually:

```
$ libxslt-1.1.28  ./configure
checking build system type... ./config.guess: line 142: /usr/sbin/sizer: No such file or directory
./config.guess: line 142: /usr/sbin/sizer: No such file or directory
./config.guess: line 142: /usr/sbin/psrinfo: No such file or directory
./config.guess: line 142: /usr/bin/uname: No such file or directory
/usr/bin/arch: invalid option -- 'k'
Try '/usr/bin/arch --help' for more information.
./config.guess: command substitution: line 142: unexpected EOF while looking for matching `''
./config.guess: command substitution: line 144: syntax error: unexpected end of file
./config.guess: command substitution: line 143: syntax error near unexpected token `;;'
./config.guess: command substitution: line 143: `exit ;;'
./config.guess: command substitution: line 144: unexpected EOF while looking for matching `"'
./config.guess: command substitution: line 145: syntax error: unexpected end of file
./config.guess: line 142: bad substitution: no closing "`" in `" in
    sun3)
echo m68k-sun-sunos${UNAME_RELEASE}
```

**Comment by e2** [comment]
The PR is here: https://github.com/sparklemotion/nokogiri/pull/1403

But I wonder if there's a way to preserve the tabs during patching - might help future problems.

**Comment by e2** [comment]
Ok, the real fix is here: https://github.com/flavorjones/mini_portile/pull/67

**Comment by larskanis** [comment]
Closing this, since it's a mini_portile issue: https://github.com/flavorjones/mini_portile/pull/67

**Comment by flavorjones** [comment]
mini_portile2 2.1.0 contains the PR mentioned above; and this dependency has been added to Nokogiri master. I expect this to be fixed in v1.6.8 final when it drops

**Comment by e2** [comment]
Thanks!
